overlord/snapstate/handlers.go

```diff
@@ -792,36 +798,29 @@ func (m *SnapManager) doDiscardSnap(t *state.Task, _ *tomb.Tomb) error {
 		}
 	}
 
-	pb := &TaskProgressAdapter{task: t}
+	pb := NewTaskProgressAdapterLocked(t)
 	typ, err := snapst.Type()
 	if err != nil {
 		return err
 	}
 	err = m.backend.RemoveSnapFiles(snapsup.placeInfo(), typ, pb)
```

Comment:
This is just the snap file itself, right?  Let's just make sure we're not blocking on waiting for the snap data to be removed, as that might be a huge tree with GBs of data.